index.js

```diff
@@ -1,8 +1,142 @@
 const express = require('express');
+const crypto = require('crypto');
+var exec = require('child_process').execFile;
 const app = express();
+var formidable = require('formidable');
+var fs = require('fs');
+var archiver = require('archiver')
+app.use(express.json())
+
+app.get('/v1/autodock', (req, res) => {
+    if (!req.query.jobId) {
+        res.status(400);
+        return res.send('Missing required parameter: jobId');
+    }
+    if (!fs.existsSync(__dirname + '/uploads/')) {
+        fs.mkdirSync(__dirname + '/uploads/');
+    }
+    uploadDirectory = __dirname + '/uploads/' + req.query.jobId;
+    if (!fs.existsSync(uploadDirectory)) {
+        res.status(400);
+        return res.send('No job with that ID.');
+    }
+    if (!fs.existsSync(uploadDirectory + '/ligand_out.pdbqt')) {
+        res.status(200);
+        return res.send('Job still processing.');
+    }
+    var outputPath = uploadDirectory + '/output.zip';
+    var output = fs.createWriteStream(outputPath);
+    var archive = archiver('zip', {
+        zlib: { level: 9 }
+    })
+    output.on('close', function () {
+        var stat = fs.statSync(outputPath);
+        res.writeHead(200, {
+            'Content-Type': 'application/zip',
+            'Content-Length': stat.size
+        });
+        var readStream = fs.createReadStream(outputPath);
+        readStream.pipe(res)
+    })
+    archive.on('error', function (err) {
+        res.status(400);
```

Comment:
I believe this qualifies as an internal rather than an external error. I think we should use status code 500 for this.